backend/log/context_grpc.go

```diff
@@ -0,0 +1,55 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	logParamsCtxMetadataKey = "loggerParamsCtxMetadata"
+	logParamSeparator       = ":"
+)
+
+// WithContextualAttributesForOutgoingContext will append the given key/value log parameters to the outgoing context.
+func WithContextualAttributesForOutgoingContext(ctx context.Context, logParams []any) context.Context {
+	if len(logParams) == 0 || len(logParams)%2 != 0 {
```

Comment:
it could be confusing to do nothing if by mistake someone calls this with an uneven set of logParams. What if `len(logParams)%2 != 0` you add an empty param at the end? (That's how the logger works, right?)

Reply:
Yeah I can see that it's quite unforgiving, but I'm aiming for simplicity here. 

>you add an empty param at the end? (That's how the logger works, right?)

Usually what I see is the `EXTRA_VALUE_AT_END` - which can be a bit strange/unexpected.

Reply:
yeah, the problem is that if someone hits this, it would be difficult to know what's going on since no error or anything is returned but up to you if you want to leave that for later.